processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -374,12 +374,23 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	return tags
 }
 
-func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) map[string]*Container {
-	containers := map[string]*Container{}
-
-	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag {
+func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) PodContainers {
+	containers := PodContainers{
+		ByID:   map[string]*Container{},
+		ByName: map[string]*Container{},
+	}
+	if !needContainerAttributes(c.Rules) {
+		return containers
+	}
+	newContainer := func(name string) *Container {
+		if c.Rules.ContainerName {
+			return &Container{Name: name}
+		}
+		return &Container{}
+	}
+	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag || c.Rules.ContainerName {
```

Comment:
Do we really need to go over this loop if `c.Rules.ContainerName`? I think we can avoid that and assign this in the second loop which we have to go through anyway